src/model/attributes/relations/BelongsTo.ts

```diff
@@ -85,20 +85,28 @@ export class BelongsTo extends Relation {
    * Match the eagerly loaded results to their respective parents.
    */
   match(relation: string, models: Collection, query: Query): void {
-    const results = query.get()
-    const dictionary = results.reduce<Record<string, Model>>((dic, result) => {
-      dic[result[this.ownerKey]] = result
-
-      return dic
-    }, {})
+    const dictionary = this.buildDictionary(query.get())
 
     models.forEach((model) => {
-      dictionary[model[this.foreignKey]]
-        ? model.$setRelation(relation, dictionary[model[this.foreignKey]])
+      const key = model[this.foreignKey]
+
+      dictionary[key]
+        ? model.$setRelation(relation, dictionary[key])
         : model.$setRelation(relation, null)
     })
   }
 
+  /**
+   * Build model dictionary keyed by relation's parent key.
+   */
+  protected buildDictionary(models: Collection): Record<string, Model> {
+    return models.reduce<Record<string, Model>>((dictionary, model) => {
+      dictionary[model[this.ownerKey]] = model
+
+      return dictionary
+    }, {})
+  }
```

Comment:
There's a recurring theme with this type of dictionary that might warrant a shorthand method equivalent to the `mapToDictionary` on the `Relation` object. Though, I'm not sure what it'd be called to distinguish a dictionary of  Model (singular) and Models (plural).

Reply:
Yeah maybe. inverse relationship is a bit special so 👀  But yeah we could create a shared function and share it. We could refactor this later.